components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/util/SecretUtils.java

```diff
@@ -0,0 +1,106 @@
+     * SecretConstants.ErrorMessages object when no exception is thrown.
+     *
+     * @param error SecretConstants.ErrorMessages.
+     * @param data  data to replace if message needs to be replaced.
+     * @return SecretManagementClientException.
+     */
+    public static SecretManagementClientException handleClientException(SecretConstants.ErrorMessages error,
+                                                                        String... data) {
+
+        String message = populateMessageWithData(error, data);
+        return new SecretManagementClientException(message, error.getCode());
+    }
+
+    public static SecretManagementClientException handleClientException(SecretConstants.ErrorMessages error,
+                                                                        String data, Throwable e) {
+
+        String message = populateMessageWithData(error, data);
+        return new SecretManagementClientException(message, error.getCode(), e);
+    }
+
+    /**
+     * This method can be used to generate a SecretManagementServerException from
+     * SecretConstants.ErrorMessages object when no exception is thrown.
+     *
+     * @param error SecretConstants.ErrorMessages.
+     * @param data  data to replace if message needs to be replaced.
+     * @return SecretManagementServerException.
+     */
+    public static SecretManagementServerException handleServerException(SecretConstants.ErrorMessages error,
+                                                                        String data) {
+
+        String message = populateMessageWithData(error, data);
+        return new SecretManagementServerException(message, error.getCode());
+    }
+
+    public static SecretManagementServerException handleServerException(SecretConstants.ErrorMessages error,
+                                                                        String data, Throwable e) {
+
+        String message = populateMessageWithData(error, data);
+        return new SecretManagementServerException(message, error.getCode(), e);
+    }
+
+    public static SecretManagementServerException handleServerException(
+            SecretConstants.ErrorMessages error, Throwable e) {
+
+        String message = populateMessageWithData(error);
+        return new SecretManagementServerException(message, error.getCode(), e);
+    }
+
+    public static String generateUniqueID() {
+
+        return UUID.randomUUID().toString();
+    }
+
+    private static String populateMessageWithData(SecretConstants.ErrorMessages error, String... data) {
+
+        String message;
+        if (data != null && data.length != 0) {
+            message = String.format(error.getMessage(), data);
+        } else {
+            message = error.getMessage();
+        }
+        return message;
+    }
+
+    private static String populateMessageWithData(SecretConstants.ErrorMessages error) {
+
+        return error.getMessage();
+    }
+
```

Comment:
Unnecessary newline

Reply:
Addressed with the commit [4191a12](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/4191a12ce26a59b9a12028b142a6964aeb4ddd2a)